### ISSUE TITLE
How to get the face texture?

### ISSUE THREAD

**Issue by sunshinewhy** [ascust/3DMM-Fitting-Pytorch#16]
Thank you for your awesome work!
After running your project, I get the reconstructed face image and *.obj file, can you tell me how to get the rendered image just like these?
![2](https://user-images.githubusercontent.com/43949229/143797116-6288af7f-bd11-409e-867f-6db26585088a.png)

**Comment by ascust** [comment]
Please have a look at this line:
https://github.com/ascust/3DMM-Fitting-Pytorch/blob/8ca87fc3ce42555d376dc505858e96536068896d/fit_single_img.py#L118
I guess this is what you are looking for. It is basically the output from the renderer of Pytorch3D, which is a 4 channel image (the forth channel is a mask). So simply convert it into a uint8 array and save it to an image type.

**Comment by sunshinewhy** [comment]
> Please have a look at this line:
> 
> https://github.com/ascust/3DMM-Fitting-Pytorch/blob/8ca87fc3ce42555d376dc505858e96536068896d/fit_single_img.py#L118
> 
> 
> I guess this is what you are looking for. It is basically the output from the renderer of Pytorch3D, which is a 4 channel image (the forth channel is a mask). So simply convert it into a uint8 array and save it to an image type.

Yes, you are right.  You provide some images in data folder,  and I want to know if I replace these images, should I retrain the model?

**Comment by ascust** [comment]
@sunshinewhy those are some sample images. It has nothing to do with model training. All the code does is "given an image or a video, the code is trying to find a set of parameters to describe the reconstructed image or the video."

**Comment by sunshinewhy** [comment]
> @sunshinewhy those are some sample images. It has nothing to do with model training. All the code does is "given an image or a video, the code is trying to find a set of parameters to describe the reconstructed image or the video."

Nice, thanks a lot!

**Comment by sunshinewhy** [comment]
I use one RTX3080 GPU to render the image, each image cost about 35s (including 4s no-rigid render and 33s rigid render), I want to reduce the time. I find you provide the hyperparameters in the file, could you tell me how to set them,  I will be grateful.

**Comment by ascust** [comment]
@sunshinewhy In the current version, I only use Adam as the optimizer to iteratively update variables, which might not be a fast way. To speed things up, one could replace the optimization method with some faster ones. I have not dug into it yet.

**Comment by sunshinewhy** [comment]
@ascust Okay, I reset the hyperparameters, it will save the time, but the results not good, maybe I should try other methods.
